exercise3.js

```diff
@@ -86,50 +114,106 @@
 const numbers = [2,23,1,2,1,1,1,2,2.5,20,200,2000,,{k:1},20000,19999,1878,140,23,4,"sk",true,true,"true-dat","nice","one","two","three","3","tea",[]];
 
 function maxNumber(numbers) {
-    //your code...
-}
+  //your code...
+  const truthys = numbers.filter(el => el);
+  const primitives = truthys.filter(n => typeof n != 'object');
+  const noBools = primitives.filter(p => typeof p != 'boolean');
+  const numStrings = ["zero","one","two","three"];
+  const numWithStrings = noBools.map(nb => numStrings.includes(nb) ? numStrings.indexOf(nb) : nb);
+  const numWithStrings2 = numWithStrings;
+  numWithStrings2[24] = Number(numWithStrings[24]);
+  const nums = numWithStrings2.filter(n => typeof n != 'string');
+
+  return Math.max(...nums);
+};
 
 // 7.b -Write a function that sorts the given numbers array.  Allow the function to sort the array in descending order
 
 function sortNums(numbers,desc=false) {
-    //your code...
+  //your code...
+  return numbers.sort().reverse();
 };
 
 
 
-
 /************************************************************* */
 // Problem 8:
 //add an example of at least 5 JavaScript data types to the given mapObj.  The key is the example data type, and the value is the name of the data type.  An object data type has already been set as the 1st key / val pair.
 
 const mapObj = new Map();
 mapObj.set({company : "TEKsystems"},"object");
-
-
-console.log(mapObj.has({company : "TEKsystems"}));  
+Object.defineProperty(mapObj, 2, {
+    value: 'number',
+    enumerable: true
+});
+Object.defineProperty(mapObj, 'Hello', {
+    value: 'string',
+    enumerable: true
+});
+Object.defineProperty(mapObj, true, {
+    value: 'boolean',
+    enumerable: true
+});
+Object.defineProperty(mapObj, [1,2,3], {
+    value: 'array',
+    enumerable: true
+});
+Object.defineProperty(mapObj, function() {}, {
+    value: 'function',
+    enumerable: true
+});
+
+console.log(mapObj.has({company : "TEKsystems"}));
+console.log("When you set and object to another object key, the key object is stringified to [object Object]")
```

Comment:
The answer has to do with objects being referenced in memory.   If they are stringified as keys, then the mapObj() should have the key `{company : TEKsystems}`.  but it returns false. 
the Map() stores any data type as a key.   if you create an object literal, another object literal with the same key / val pairs are living in another memory location, so the two references are not equal.  Hope this makes sense.